virttest/virt_vm.py

```diff
@@ -935,10 +935,11 @@ def _get_address():
 
         ipaddr = utils_misc.wait_for(_get_address, timeout, step=interval)
         if not ipaddr:
-            # Read guest address via serial console and update VM address
-            # cache to avoid get out-dated address.
-            utils_net.update_mac_ip_address(self, timeout)
-            ipaddr = self.get_address(nic_index, ip_version)
+            # obatining ip address from virsh-net-dhcp-leases command
+            mac = self.get_mac_address(nic_index).lower()
+            ipaddr = utils_net.obtain_guest_ip_from_dhcp_leases(mac)
+            # updating cache with the latest ip address value
+            self.address_cache[mac] = ipaddr
```

Comment:
Hi @TasmiyaNalatwad  Could you please help redesign this part? Because of the current usage let all the releated
nic_hotplug cases failed which blocking my test.

Reply:
Hi @yanglei-rh there is patch already shared by @luckyh #3989 to revert this patch, Which would unblock your tests. Also i will share the new patch with the approach we had discussed in above comments soon.

Thank you.

Reply:
Got it, thanks a lot.